### ISSUE TITLE
Check precision of incomplete gamma 

### ISSUE THREAD

**Issue by Shimuuar** [bos/math-functions#41]
Following comment in incomplete gamma implementation does sound very reassuring

```haskell
    -- For very large `p' normal approximation gives <1e-10 error
    | p >= 2e5           = norm (3 * sqrt p * ((x/p) ** (1/3) + 1/(9*p) - 1))
```

**Comment by Shimuuar** [comment]
Implementation of incomplete gamma provides about 10 signiticant digits. Sometimes worse

**Comment by Shimuuar** [comment]
Mostly fixed in dd10135d14a0da2a8b66da275de569490bb32442

There's still room for improvements but these will be filed as separate issues.
